src/runtime/SmallValue.h

```diff
@@ -340,35 +297,18 @@ class SmallValue {
         if (from.isPointerValue()) {
 #ifdef ESCARGOT_32
             ASSERT(!from.isEmpty());
-            m_data.payload = (intptr_t)from.asPointerValue();
-#else
-            if (from.isEmpty()) {
-                m_data.payload = (intptr_t)(smallValueEmpty);
-            } else {
-                m_data.payload = (intptr_t)from.asPointerValue();
-            }
-
 #endif
+            m_data.payload = (intptr_t)from.asPointerValue();
         } else {
             int32_t i32;
             if (from.isInt32() && SmallValueImpl::PlatformSmiTagging::IsValidSmi(i32 = from.asInt32())) {
                 m_data.payload = SmallValueImpl::PlatformSmiTagging::IntToSmi(i32);
```

Comment:
I'm wondering that validity check & conversion to small integer value is really necessary for 64 bit platform? I think that every 32 bit integer value should be represented in 64 bit value. So, it seems that just shifting 1 bit and adding 1 for integer value is enough for representing integer value in `SmallValue`.

Reply:
I am willing to change this if you think this should be in this patch (this is an old code as well). Unfortunately the purpose of some code is not entirely clear for me here (still learning the project), but I think many future optimization opportunities are available, but I would do them in separate patches rather than one big one. It is easier to track failures in the future this way.

Reply:
It may be one of optimization opportunities. We can try this issue if it has performance benefit. Thank you.